redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/CustomRedisConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package org.springframework.cloud.stream.module.redis;
+
+import java.net.UnknownHostException;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import redis.clients.jedis.Jedis;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration;
+import org.springframework.boot.autoconfigure.redis.RedisProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+
+/**
+ * Creates a dedicated RedisConnectionFactory (that is, different from the spring-cloud-stream one).
+ *
+ * @author Eric Bottard
+ */
+@Configuration
+@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
+@EnableConfigurationProperties
+public class CustomRedisConfiguration extends RedisAutoConfiguration {
```

Comment:
I don't think there is any issue here.  Using another prefix is handled by adding @ConfigurationProperties at the bean level.  The only thing that needs to be distinguished ATM is the redis connection factory used for the sink since the standard namespace 'spring.redis' is being used for the binding.  I think actually making the binding more isolated with a different namespace would be a better way so that frees up them module to be more easily developed - ATM this would effect redis and rabbit, but eventually include kafka and probably the datasource for the batch db.